common/navbar.js

```diff
@@ -59,7 +59,24 @@
             var parentNewTab = obj.newTab;
             // template the url
             // TODO: This is done here to prevent writing a recursive function (again) in `setConfigJSON()`
-            if (obj.url && isCatalogDefined(catalogId)) obj.url = ERMrest.renderHandlebarsTemplate(obj.url, null, {id: catalogId});
+            if (obj.url && isCatalogDefined(catalogId)) {
+                obj.url = ERMrest.renderHandlebarsTemplate(obj.url, null, {id: catalogId});
+                // only append pcid/ppid if link is to the same origin
+                // same origin urls will be relative, check for http instead(?)
+                // if http, make sure not same origin
+                if (obj.url.indexOf("http") == -1 || obj.url.indexOf($window.location.origin) != -1) {
+                    var paramChar = "";
+                    if (obj.url.lastIndexOf("?") !== -1) {
+                        // already a `?` in the url
+                        paramChar += '&';
+                    } else {
+                        paramChar += '?';
+                    }
+                    // pcid should always be navbar
+                    // ppid should be the pid for the current page
+                    obj.url += paramChar + "pcid=navbar&ppid=" + $window.dcctx.pid;
```

Comment:
It's better if we stick with `ConfigUtils.getContextJSON()` instead of directly using `$window` object. So if later we decided to change this, we just have to change one function.